Objects/src/GenParticle.cc

```diff
@@ -15,6 +15,7 @@ GenParticle::GenParticle(const GenParticle & genParticle)
     , genParticle.status_
     , genParticle.statusFlags_
     , genParticle.genPartFlav_
+    , genParticle.genHadronFlav_
```

Comment:
Please revert the changes in `GenParticle` class. Instances of `GenJet` and `GenParticle` are objects of their own right. Just propagate `partonFlavor` of `GenJet` to `GenParticle`.